api/coreservice.go

```diff
@@ -520,15 +518,6 @@ func (core *coreService) SuggestGasPrice() (uint64, error) {
 	return core.gs.SuggestGasPrice()
 }
 
-// EstimateGasForAction estimates gas for action
-func (core *coreService) EstimateGasForAction(in *iotextypes.Action) (uint64, error) {
```

Comment:
i think what the issue #3247 asked, is to directly call `sf.SimulateExecution` here, instead of pass it to gs, and call `gs.EstimateGasForAction`

all the changes should be in this func, there's no need to remove this func, and no need to change `grpcserver.go`

Reply:
ok.